src/js/components/Diagram/stories/Progressing.js

```diff
@@ -28,57 +28,58 @@ const connection = (fromTarget, toTarget, { color, ...rest } = {}) => ({
 });
 
 const fullTopRow = [1, 2, 3];
-class SimpleDiagram extends React.Component {
-  state = { topRow: fullTopRow.slice(0, 1) };
 
-  componentDidMount() {
-    this.timer = setInterval(() => {
-      const { topRow } = this.state;
-      this.setState({
-        topRow: fullTopRow.slice(
-          0,
-          topRow.length < fullTopRow.length ? topRow.length + 1 : 1,
-        ),
-      });
+const SimpleDiagram = () => {
+  const reducer = topRow => {
+    const sliceEnd = topRow.length < fullTopRow.length ? topRow.length + 1 : 1;
+    return fullTopRow.slice(0, sliceEnd);
+  };
+
+  const [topRow, dispatch] = useReducer(reducer, fullTopRow.slice(0, 1));
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      dispatch();
     }, 2000);
+    return () => clearInterval(timer);
+  }, [dispatch]);
+
+  const connections = [connection('1', '5', { color: 'accent-2' })];
+
+  if (topRow.length >= 2) {
+    connections.push(
+      connection('1', '2', { color: 'accent-1', anchor: 'horizontal' }),
+    );
   }
 
-  render() {
-    const { topRow } = this.state;
-    const connections = [connection('1', '5', { color: 'accent-2' })];
-    if (topRow.length >= 2) {
-      connections.push(
-        connection('1', '2', { color: 'accent-1', anchor: 'horizontal' }),
-      );
-    }
-    if (topRow.length >= 3) {
-      connections.push(
-        connection('3', '5', { color: 'accent-2', anchor: 'horizontal' }),
-      );
-    }
-    return (
-      <Grommet theme={grommet}>
-        <Box align="start" pad="large">
-          <Text> Adding and removing nodes</Text>
-          <Stack>
-            <Box>
-              <Box direction="row">
-                {topRow.map(id => (
-                  <Node key={id} id={id} />
-                ))}
-              </Box>
-              <Box direction="row">
-                {[4, 5].map(id => (
-                  <Node key={id} id={id} background="neutral-2" />
-                ))}
-              </Box>
-            </Box>
-            <Diagram connections={connections} />
-          </Stack>
-        </Box>
-      </Grommet>
+  if (topRow.length >= 3) {
+    connections.push(
+      connection('3', '5', { color: 'accent-2', anchor: 'horizontal' }),
     );
   }
-}
+
+  return (
+    <Grommet theme={grommet}>
+      <Box align="start" pad="large">
+        <Text> Adding and removing nodes</Text>
+        <Stack>
+          <Box>
+            <Box direction="row">
+              {topRow.map(id => (
+                <Node key={id} id={id} />
+              ))}
+            </Box>
+            <Box direction="row">
+              {[4, 5].map(id => (
+                <Node key={id} id={id} background="neutral-2" />
+              ))}
+            </Box>
+          </Box>
+          <Diagram connections={connections} />
+        </Stack>
+      </Box>
+    </Grommet>
+  );
+};
 
 storiesOf('Diagram', module).add('Progressing', () => <SimpleDiagram />);
```

Comment:
```suggestion
storiesOf('Diagram', module).add('Progressing', () => <ProgressingDiagram />);
```
Renaming the component to use a proper name that is aligned with the story.